packages/shared/src/server/llm/fetchLLMCompletion.ts

```diff
@@ -69,8 +97,35 @@ export async function fetchLLMCompletion(
     baseURL,
     maxRetries,
     config,
+    traceParams,
   } = params;
 
+  let finalCallbacks: BaseCallbackHandler[] | undefined = callbacks ?? [];
+  let processTracedEvents: ProcessTracedEvents = () => Promise.resolve();
+
+  if (traceParams) {
+    const handler = new CallbackHandler({
+      tags: traceParams.tags,
+    });
+
+    finalCallbacks.push(handler);
+
+    processTracedEvents = async () => {
```

Comment:
marlies will be calling this function, correct? So the latency penalty occurs wherever she calls.

Reply:
correct!